pkg/controller/certificatesigningrequests/selfsigned/selfsigned.go

```diff
@@ -115,8 +115,6 @@ func (s *SelfSigned) Sign(ctx context.Context, csr *certificatesv1.CertificateSi
 		message := fmt.Sprintf("Referenced Secret %s/%s not found", resourceNamespace, secretName)
 		log.Error(err, message)
 		s.recorder.Event(csr, corev1.EventTypeWarning, "SecretNotFound", message)
-		util.CertificateSigningRequestSetFailed(csr, "SecretNotFound", message)
```

Comment:
Just to confirm, there's no such thing as Pending in the CSR API right? It's pending by virtue of it being neither issued nor failed?

Reply:
[Exactly](https://github.com/kubernetes/api/blob/fa32a3acacac7f887587d613631020889414b232/certificates/v1/types.go#L217). We _could_ add a special cert-manager `Pending` condition in future if we wanted to, but these would have no meaning to the API server and wouldn't benefit from the [same enforcement](https://github.com/kubernetes/api/blob/fa32a3acacac7f887587d613631020889414b232/certificates/v1/types.go#L240) the well known ones do.